test/testmachinery/system/shoot_update/update_test.go

```diff
@@ -110,7 +113,25 @@ func RunTest(
 		By("Update .kubernetes.version to " + kubernetesVersion + " for pool " + poolName)
 	}
 
-	Expect(f.UpdateShoot(ctx, func(shoot *gardencorev1beta1.Shoot) error {
+	var hasAutoInplaceUpdate, hasManualInplaceUpdate, hasInplaceUpdate bool
+
+	for _, worker := range f.Shoot.Spec.Provider.Workers {
+		switch ptr.Deref(worker.UpdateStrategy, "") {
+		case gardencorev1beta1.AutoInPlaceUpdate:
+			hasAutoInplaceUpdate = true
+		case gardencorev1beta1.ManualInPlaceUpdate:
+			hasManualInplaceUpdate = true
+		}
+	}
+
+	hasInplaceUpdate = hasAutoInplaceUpdate || hasManualInplaceUpdate
+
+	var nodesOfInPlaceWorkersBeforeTest sets.Set[string]
+	if hasInplaceUpdate {
+		nodesOfInPlaceWorkersBeforeTest = inplace.FindAllNodesOfInPlaceWorker(ctx, f.ShootClient.Client(), f.Shoot)
+	}
```

Comment:
```suggestion
	nodesOfInPlaceWorkersBeforeTest := inplace.FindAllNodesOfInPlaceWorker(ctx, f.ShootClient.Client(), f.Shoot)
```
The inside loop takes care of the condition already